snap/validate.go

```diff
@@ -400,6 +400,38 @@ func ValidateLayoutAll(info *Info) error {
 	}
 	sort.Strings(paths)
 
+	// Validate that each source path is not a new top-level directory
+	for pathSrc := range info.Layout {
+		cleanPathSrc := info.ExpandSnapVariables(filepath.Clean(pathSrc))
```

Comment:
This is puzzling. Having expanded the variables it still is confused. Looking at the rest of the code we seem to use `Layout.Path`J